### PR TITLE
Template: handle template features with a yaml file

### PR DESCRIPTION
**Pull request by mirpedrol**
To avoid repetitive code for every pipeline feature we want to add, this PR adds a yaml file which will be the source of true and which all the commands will use.
`nf-core pipelines create` -> to create the pipeline
`tests/data/pipeline_create_template_skip.yml` -> to test skipping all features
`create-test-lint-wf-template.yml `-> to test creating a pipeline skipping all features, running it and linting

### CODE REVIEW

File: .github/workflows/create-test-lint-wf-template.yml [hunk 1]
```diff
@@ -31,31 +31,41 @@ env:
...
+    name: Retrieve all template features
+    runs-on: ubuntu-latest
+    outputs:
+      all_features: ${{ steps.create_matrix.outputs.matrix }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: Create Matrix
+        id: create_matrix
+        run: echo "matrix=$(yq 'keys' ${{ github.workspace }}/nf_core/pipelines/create/templatefeatures.yml)" >> $GITHUB_OUTPUT
```
Comment (mashehu) [inline .github/workflows/create-test-lint-wf-template.yml:51]: need to checkout the code first I think

File: nf_core/pipelines/create/create.py [hunk 2]
```diff
@@ -81,45 +81,28 @@ def __init__(
...
-            "changelog": ["CHANGELOG.md"],
-        }
+        # format strings in features yaml
+        short_name = self.jinja_params["short_name"]
+        env = jinja2.Environment(loader=jinja2.PackageLoader("nf_core", "pipelines"), keep_trailing_newline=True)
+        features_template = env.get_template(
+            str(features_yml_path.relative_to(Path(nf_core.__file__).parent / "pipelines"))
+        )
+        rendered_features = features_template.render({"short_name": short_name})
+        self.template_features_yml = yaml.safe_load(rendered_features)
```
Comment (mashehu) [inline nf_core/pipelines/create/create.py:102]: why are we overwriting the features here?

File: nf_core/pipelines/create/create.py [hunk 3]
```diff
@@ -211,46 +194,30 @@ def obtain_jinja_params_dict(self, features_to_skip: List[str], pipeline_dir: Un
 
         Returns:
             jinja_params (dict): Dictionary of template areas to skip with values true/false.
-            skip_paths (list<str>): List of template areas which contain paths to skip.
+            skip_areas (list<str>): List of template areas which contain paths to skip.
```
Comment (mashehu) [inline nf_core/pipelines/create/create.py:197]: add them also as output type annotations to the function

File: nf_core/pipelines/create/templatefeatures.yml [hunk 4]
```diff
@@ -0,0 +1,237 @@
+github:
+  skippable_paths:
+    - ".github/"
+    - ".gitignore"
+  short_description: ""
```
Comment (mashehu) [inline nf_core/pipelines/create/templatefeatures.yml:5]: should add a short description

File: nf_core/pipelines/create/templatefeatures.yml [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  description: ""
+  help: ""
```
Comment (mashehu) [inline nf_core/pipelines/create/templatefeatures.yml:7]: should be `help_text` to make it a bit clearer

**Review by mashehu — APPROVED** [review]

File: nf_core/pipelines/create/create.py [hunk 6]
```diff
@@ -81,45 +81,28 @@ def __init__(
...
-                "assets/methods_description_template.yml",
-                "modules/nf-core/multiqc/",
-            ],
-            "changelog": ["CHANGELOG.md"],
-        }
+        # format strings in features yaml
+        short_name = self.jinja_params["short_name"]
+        env = jinja2.Environment(loader=jinja2.PackageLoader("nf_core", "pipelines"), keep_trailing_newline=True)
+        features_template = env.get_template(
+            str(features_yml_path.relative_to(Path(nf_core.__file__).parent / "pipelines"))
```
[the remark this answers is not in the document]
Comment (mirpedrol) [reply nf_core/pipelines/create/create.py:99]: the path is used here only, I think we don't need to make it a class attribute

File: nf_core/pipelines/create/utils.py [hunk 7]
```diff
@@ -244,22 +249,7 @@ def remove_hide_class(app, widget_id: str) -> None:
...
-For more information about reference genomes in nf-core pipelines,
-see the [nf-core docs](https://nf-co.re/docs/usage/reference_genomes).
-"""
-
-markdown_multiqc = """
-MultiQC is a visualization tool that generates a single HTML report summarising all samples in your project. Most of the pipeline quality control results can be visualised in the report and further statistics are available in the report data directory.
-
-The pipeline will include the MultiQC module and will have special steps which also allow the software versions to be reported in the MultiQC output for future traceability. For more information about how to use MultiQC reports, see http://multiqc.info.
-"""
+def load_features_yaml() -> dict:
```
Comment (mashehu) [inline nf_core/pipelines/create/utils.py:252]:
```suggestion
def load_features_yaml() -> Dict:
```
to make it compatible with 3.8

**Pull request merged**